### PR TITLE
BUG: Incorrect handling of rolling.cov with offset window

### LINKED ISSUE #16058: Rolling.cov with offset window is expanding not rolling
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np

idx = pd.date_range('2017-01-01', periods=24, freq='1h')
ss = pd.Series(np.arange(len(idx)), index=idx)

(ss.rolling('2h').cov() == ss.expanding().cov()).iloc[2:].any() # this should be False!!


```
#### Problem description

When using Rolling.cov with an offset window like '2h' the window is expanding and not rolling

#### Expected Output

Rolling should not be equal to expanding cov - but it is equal.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 45 Stepping 7, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.19.2
nose: 1.3.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.1
numpy: 1.11.3
scipy: 0.18.1
statsmodels: 0.8.0
xarray: None
IPython: 5.1.0
sphinx: 1.4.1
patsy: 0.4.1
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: None
tables: 3.2.2
numexpr: 2.6.1
matplotlib: 1.5.1
openpyxl: None
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: 0.9.2
lxml: None
bs4: 4.5.1
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.13
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.42.0
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by keitakurita**
- [x] closes #16058
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/test_window.py [hunk 1]
```diff
@@ -3396,6 +3396,18 @@ def test_min_periods(self):
         result = df.rolling('2s', min_periods=1).sum()
         tm.assert_frame_equal(result, expected)
 
+    def test_cov(self):
+
```
Comment (jreback) [inline pandas/tests/test_window.py]: comment with the issue

Comment (keitakurita) [inline pandas/tests/test_window.py]: fixed!

File: pandas/core/window.py [hunk 2]
```diff
@@ -996,7 +997,10 @@ def cov(self, other=None, pairwise=None, ddof=1, **kwargs):
             # only default unset
             pairwise = True if pairwise is None else pairwise
         other = self._shallow_copy(other)
-        window = self._get_window(other)
+        if self.is_freq_type:
```
Comment (jreback) [inline pandas/core/window.py]: you can do this in ``_get_window`` itself i think.

**Review by jreback — CHANGES_REQUESTED** [review]
add the test from the issue. I would prefer that instead of comparing the results, you construct the correct results for each case.

add a whatsnew note in 0.20.2 bug fixes / rolling section

File: pandas/core/window.py (hunk 2 above)
Comment (keitakurita) [inline pandas/core/window.py]: I tried changing `_get_window`, but a lot of methods (such as `var`, `median`, and `sum`) require an integer window and seem to be working fine right now. If this issue is not a problem in methods other than `cov`, changing the code only for `cov` seems to make more sense.

Comment (keitakurita) [inline pandas/core/window.py]: test for issue and whatsnew note added

Comment (jreback) [inline pandas/core/window.py]: can you add a comment here (1-line) about this

File: pandas/tests/test_window.py [hunk 3]
```diff
@@ -3801,3 +3801,20 @@ def test_non_monotonic(self):
...
         result = df2.groupby('A').rolling('4s', on='B').C.mean()
         tm.assert_series_equal(result, expected)
+
+    def test_rolling_cov_offset(self):
+        # GH16058
+
+        idx = pd.date_range('2017-01-01', periods=24, freq='1h')
+        ss = pd.Series(np.arange(len(idx)), index=idx)
+
+        result = ss.rolling('2h').cov()
```
Comment (jreback) [inline pandas/tests/test_window.py]: also test again using ``ss.rolling(2).cov()`` (will be the same for this case).

I don't know if we have any tests for a non-regular date range (Iow where freq='2' and freq='2h' are actually different.

Comment (keitakurita) [inline pandas/tests/test_window.py]: test added

**Comment by TomAugspurger** [comment]
Rebased. @jreback I think everything was addressed, if you want to verify?

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**